skimage/exposure/exposure.py

```diff
@@ -34,9 +76,17 @@ def histogram(image, nbins=256):
     ----------
     image : array
         Input image.
-    nbins : int
+    nbins : int, optional
         Number of bins used to calculate histogram. This value is ignored for
         integer arrays.
+    source_range : string or tuple, optional
+        'image' determines the range from the input image.
+        'dtype' determines the range from the expected range of the images
+        of that data type.
+        `None` corresponds to 'image'. This behavior will change in
+        version 0.16 and set default to 'dtype'.
```

Comment:
Is this a good idea?  See e.g.

https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html

Reply:
All this PR is based on the fact that, in practice, we find this behavior unexpected and not so practical. If you ignore np, you would probably guess that a histogram function would return the values in N bins from 0 to max of the range, not something cropped on the min(img) and max(img)

Reply:
I don't think that would be my intuition---almost all histograms I've seen only cover the data range.